### PR TITLE
CTX-5783: Mypy Linter Action

### PR DESCRIPTION
**Pull request by VukManojlovic**

### CODE REVIEW

**Review by dule1322 — CHANGES_REQUESTED** [review]
Can you check which libraries added to mypy config file have types and remove them from there?

File: tasks/bio-bodysite-prediction/src/cache_json.py [hunk 1]
```diff
@@ -18,13 +18,16 @@ def getJsonName(datasetName: str, sampleOrigin: list[str], sequencingTechnique:
 
     suffix = f"{origins}-{techniques}"
 
-    return hashCacheName(datasetName, suffix)
+    return hashCacheName(datasetName, suffix)[:20]
```
[the remark this answers is not in the document]
Comment (VukManojlovic) [reply tasks/bio-bodysite-prediction/src/cache_json.py:21]: Not sure. Will change it back for now.

File: tasks/bio-bodysite-prediction/src/load_data.py [hunk 2]
```diff
@@ -334,7 +334,7 @@ def loadDataAtlas(
         return loadMatrixCache(cacheNameMatrix, validate)
 
     cacheNameJson = getJsonName(
-        dataset.name,
+        dataset.name[:42],
```
[the remark this answers is not in the document]
Comment (VukManojlovic) [reply tasks/bio-bodysite-prediction/src/load_data.py:337]: Not sure. Will change it back for now.v

File: tasks/qiime-import/mypy.ini [hunk 3]
```diff
@@ -17,3 +17,12 @@ disallow_incomplete_defs = True
...
 allow_redefinition = False
+
+
+# Per-module options:
+# https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
+[mypy-cchardet.*]
+ignore_missing_imports = True
+
+[mypy-pandas.*]
+ignore_missing_imports = True
```
Comment (dule1322) [inline tasks/qiime-import/mypy.ini:28]: pandas has package called `pandas-stubs` which gets installed by running `mypy --install-types` so it should not be present in this file

File: tasks/bio-read-quality/.mypy.ini [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+disallow_untyped_defs = True
+disallow_incomplete_defs = True
+no_implicit_optional = True
+strict_optional = True
+allow_redefinition = False
+
+
+# Per-module options:
+# https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
+[mypy-matplotlib.*]
```
Comment (dule1322) [inline tasks/bio-read-quality/.mypy.ini:23]: matplotlib has package called `matplotlib-stubs` which gets installed by running `mypy --install-types` so it should not be present in this file

**Review by dule1322 — CHANGES_REQUESTED** [review]

File: tasks/llm-text-processing/main.py [hunk 5]
```diff
@@ -105,7 +107,7 @@ def main() -> None:
 
         logging.info(f">> [LLMTextProcessing] Prompt: {prompt}")
 
-        responseContent = response["message"]["content"]
+        responseContent = response["message"]["content"]  # type: ignore[index]
```
[the remark this answers is not in the document]
Comment (dule1322) [reply tasks/llm-text-processing/main.py]: It is better as it keeps the code using expected types

**Review by dule1322 — APPROVED** [review]

**Pull request merged**